jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/form/provider/InMemoryFormSkeletonProvider.java

```diff
@@ -0,0 +1,181 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.commons.lang.StringEscapeUtils;
+import org.jbpm.kie.services.impl.model.ProcessAssetDesc;
+import org.jbpm.services.api.model.ProcessDefinition;
+import org.kie.api.task.model.Task;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class InMemoryFormSkeletonProvider extends AbstractFormProvider {
+    
+    public static final String NODE_FORM = "form";
+    public static final String NODE_FIELD = "field";
+    public static final String NODE_PROPERTY = "property";
+    public static final String NODE_DATA_HOLDER = "dataHolder";
+
+    public static final String ATTR_NAME = "name";
+    public static final String ATTR_VALUE = "value";
+    public static final List<String> ATTR_LANG_NAMES = Arrays.asList("label", "errorMessage", "title");
+
+    @Override
+    public String render(String name, ProcessDefinition process, Map<String, Object> renderContext) {
+        if (!(process instanceof ProcessAssetDesc)) {
+            return null;
+        }
+
+        String templateString = formManagerService.getFormByKey(process.getDeploymentId(), process.getId());
+        if (templateString == null) {
+            templateString = formManagerService.getFormByKey(process.getDeploymentId(), process.getId() + getFormSuffix());
+        }
+
+        if (templateString == null || templateString.isEmpty()) {
+            return null;
+        } else {
+            String lang = (String) renderContext.get("lang");
+            templateString = filterXML(templateString, lang, null);
+            return templateString;
+        }
+    }
+
+    @Override
+    public String render(String name, Task task, ProcessDefinition process, Map<String, Object> renderContext) {
+        if (task == null) return null;
+
+        String lookupName = getTaskFormName( task );
+
+        if ( lookupName == null || lookupName.isEmpty()) return null;
+
+        String templateString = formManagerService.getFormByKey(task.getTaskData().getDeploymentId(), lookupName);
+
+        if (templateString == null || templateString.isEmpty()) {
+            return null;
+        } else {
+            Map inputs = new HashMap();
+            Map m = (Map) renderContext.get("inputs");
+            if (m != null) {
+                inputs.putAll(m);
+            }
+            
+            String lang = (String) renderContext.get("lang");
+            
+            templateString = filterXML(templateString, lang, inputs);
+            
+            return templateString;
+        }
+    }
+    
+    private String filterXML(String document, String lang, Map inputs) {
```

Comment:
there two types of filtering - filtering of the inputBinding where we replace the location (user/name) for its value (Ivo) .... the second is language filtering to situations when we need a form in a specific language, so all titles, names, labels, error messages, ... are presented only in this specific language because I believe there is no need to send all language strings to the client.